### DISCUSSION TITLE
Is it possible to make Pixorama server stateless?

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#867] [Q&A] [answered]
Just store all "States" in Database, and use API to check latest status of the drawing or user session, to make Pixorama server stateless, is this possible? If yes, what problems we could incur?

**Comment by vlidholt** [comment, marked as answer]
It is possible to do what you describe and that may be the preferred way to do some type of apps. However, in the case of Pixorama, it would be very inefficient/expensive to write every single change to the database. Imagine thousands of users clicking away in the interface, every pixel clicked would need to be a write to the database.

This type of application is probably a couple orders of magnitude more efficient as stateful.
